### ISSUE TITLE
Extending from Error doesn't work correctly under tsc 1.6.2

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#5069]
Test case:

myError.ts
```typescript
class MyError extends Error {
	constructor(message: string) {
		super(message);
	}
}

let error = new MyError('Error Message');
console.log(error.message);
```
tsconfig.json
```json
{
    "compilerOptions": {
        "target": "ES5",
        "module": "commonjs"
    }
}
```
compile and run using node:

node myError.js

Observe: nothing is printed to the console. Debugging it shows that the message passed to super is not recorded. Something seems very strange with the prototype chain in this case.

I was under the impression from https://github.com/Microsoft/TypeScript/pull/3516 that this should work now even if generating ES5 code. If not possible for ES5 then the compiler should produce an error.

**Comment by mhegazy** [comment]
The issue is with how these built in types are hand eked by the engine. They are extensible as per the es6 spec, but do not think the engines are there yet.

Here is some related discussion https://github.com/Microsoft/TypeScript/issues/1168#issuecomment-107833988
